File: java/src/main/java/org/example/Alarm.java
package org.example;

public class Alarm {

    private static final double LOW_PRESSURE_THRESHOLD = 17;
    private static final double HIGH_PRESSURE_THRESHOLD = 21;

    Sensor sensor = new Sensor();

    boolean alarmOn = false;
    private long alarmCount = 0;


    public void check()
    {
        double psiPressureValue = sensor.popNextPressurePsiValue();

        if (psiPressureValue < LOW_PRESSURE_THRESHOLD || HIGH_PRESSURE_THRESHOLD < psiPressureValue)
        {
            alarmOn = true;
            alarmCount += 1;
        }
    }

    public boolean isAlarmOn()
    {
       return alarmOn;
    }
}

File: kotlin/src/main/kotlin/org/example/Alarm.kt
package org.example

private const val LOW_PRESSURE_THRESHOLD = 17.0
private const val HIGH_PRESSURE_THRESHOLD = 21.0

class Alarm {

    private val sensor = Sensor()
    private var alarmOn = false
    private var alarmCount = 0

    fun check() {
        val psiPressureValue = sensor.popNextPressurePsiValue()

        if (psiPressureValue < LOW_PRESSURE_THRESHOLD || psiPressureValue > HIGH_PRESSURE_THRESHOLD) {
            alarmOn = true
            alarmCount++
        }
    }

    fun isAlarmOn(): Boolean {
        return alarmOn
    }
}